facilitator/src/manifest.rs

```diff
@@ -0,0 +1,336 @@
+    batch_signing_public_keys: HashMap<String, BatchSigningPublicKey>,
+    /// Certificates containing public keys that should be used to encrypt
+    /// ingestion share packets intended for this data share processor.
+    #[serde(rename = "packet-encryption-certificates")]
+    packet_encryption_certificates: HashMap<String, PacketEncryptionCertificate>,
+}
+
+impl SpecificManifest {
+    fn from_https(base_path: &str, peer_name: &str) -> Result<SpecificManifest> {
+        if !base_path.starts_with("https://") {
+            return Err(anyhow!("Manifest must be fetched over HTTPS"));
+        }
+        let manifest_url = format!("{}/{}/specific-manifest.json", base_path, peer_name);
+        let response = ureq::get(&manifest_url)
+            // By default, ureq will wait forever to connect or
+            // read.
+            .timeout_connect(10_000) // ten seconds
+            .timeout_read(10_000) // ten seconds
+            .call();
+        if response.error() {
+            return Err(anyhow!("failed to fetch specific manifest: {:?}", response));
+        }
+        SpecificManifest::from_reader(response.into_reader())
+    }
+
+    fn from_file(path: &Path) -> Result<SpecificManifest> {
+        SpecificManifest::from_reader(File::open(path).context("failed to open manifest file")?)
+    }
+
+    fn from_reader<R: Read>(reader: R) -> Result<SpecificManifest> {
+        let manifest: SpecificManifest =
+            from_reader(reader).context("failed to decode JSON specific manifest")?;
+        if manifest.format != 0 {
+            return Err(anyhow!("unsupported manifest format {}", manifest.format));
+        }
+        Ok(manifest)
+    }
+
+    /// Returns the ECDSA P256 public key corresponding to the provided key
+    /// identifier, if it exists in the manifest.
+    fn batch_signing_public_key(&self, identifier: &str) -> Result<UnparsedPublicKey<Vec<u8>>> {
+        // No Rust crate that we have found gives us an easy way to parse PKIX
+        // SubjectPublicKeyInfo structures to get at the public key which can
+        // then be used in ring::signature. Since we know the keys we deal with
+        // should always be ECDSA P256, we can instead check that the binary
+        // blob inside the PEM has the expected prefix for this kind of key in
+        // this kind of encoding, as suggested in this GitHub issue on ring:
+        // https://github.com/briansmith/ring/issues/881
+        let key = self
+            .batch_signing_public_keys
+            .get(identifier)
+            .context(format!("no value for key {}", identifier))?;
+
+        let pem = pem::parse(&key.public_key)
+            .context(format!("failed to parse key entry {} as PEM", identifier))?;
+        if pem.tag != "PUBLIC KEY" {
+            return Err(anyhow!(
+                "key for identifier {} is not a PEM encoded public key"
+            ));
+        }
+        if pem.contents.len() < ECDSA_P256_SPKI_PREFIX.len() {
```

Comment:
I did some checking and figured out that the size of the signature is alway 520 bits (or 65 bytes):

```py
>>> import base64
>>> prefix = bytes([0x30, 0x59, 0x30, 0x13, 0x06, 0x07, 0x2a, 0x86, 0x48, 0xce, 0x3d, 0x02, 0x01, 0x06, 0x08, 0x2a, 0x86, 0x48, 0xce, 0x3d, 0x03, 0x01, 0x07, 0x03, 0x42, 0x00,])
>>> base64.b64encode(prefix + bytes([0xFF]*65))
b'MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgD//////////////////////////////////////////////////////////////////////////////////////w=='
```

Which can be [fed into asn1js](https://lapo.it/asn1js/#MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgD______________________________________________________________________________________w) to verify it parses. asn1js also warns if there's excess data at the end, so this is the right number of bytes.

So this way you can also check that the entire SPKI has a length of 91 bytes, and the signature 65 bytes, to be a bit more strict.